service/lib/dinstaller/questions_manager.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module DInstaller
+  # Manager for questions
+  #
+  # Allows to configure callbacks with the actions to perform when adding, deleting or waiting for
+  # questions.
+  class QuestionsManager
+    # @return [Array<Question>]
+    attr_reader :questions
+
+    # Constructor
+    def initialize
+      @questions = []
+      @on_add_callbacks = []
+      @on_delete_callbacks = []
+      @on_wait_callbacks = []
+    end
+
+    # Adds a question
+    #
+    # Callbacks are called after adding the question, see {#on_add}.
+    #
+    # @param question [Question]
+    def add(question)
+      questions << question
+      on_add_callbacks.each { |c| c.call(question) }
+
+      questions
+    end
+
+    # Deletes the given question
+    #
+    # Callbacks are called after deleting the question, see {#on_delete}.
+    #
+    # @param question [Question]
+    def delete(question)
+      return unless questions.include?(question)
+
+      questions.delete(question)
+      on_delete_callbacks.each { |c| c.call(question) }
+
+      questions
+    end
+
+    # Waits until all questions are answered
+    #
+    # Callbacks are periodically called while waiting, see {#on_wait}.
+    def wait
+      loop do
+        on_wait_callbacks.each(&:call)
+        sleep(0.1)
+        break if questions_answered?
+      end
+    end
+
+    # Registers a callback to be called when a new question is added
+    #
+    # @param block [Proc]
```

Comment:
I kind of missing here block definition. Because on_add and on_delete have parameter and on_wait does not have it.

Reply:
btw usually it is not documented as block, but using yield tag together with its parts for param and return https://www.rubydoc.info/gems/yard/file/docs/Tags.md#yieldparam